include/openssl/ocsp.h

```diff
@@ -64,6 +72,15 @@ DECLARE_ASN1_FUNCTIONS(OCSP_RESPONSE)
 DECLARE_ASN1_FUNCTIONS(OCSP_CERTID)
 DECLARE_ASN1_FUNCTIONS(OCSP_REQUEST)
 
+// d2i_OCSP_RESPONSE_bio parses an OCSP response in DER format from a BIO object
+// and converts it into an |OCSP_RESPONSE|.
+OPENSSL_EXPORT OCSP_RESPONSE *d2i_OCSP_RESPONSE_bio(BIO *bp,
+                                                    OCSP_RESPONSE **presp);
+
+// i2d_OCSP_RESPONSE_bio parses an |OCSP_RESPONSE| structure and writes the OCSP
+// response in DER format into a BIO object.
+OPENSSL_EXPORT int i2d_OCSP_RESPONSE_bio(BIO *bp, OCSP_RESPONSE *presp);
```

Comment:
How does ASN1_item_d2i_bio work? Does it allocate the response for you, or do you need to pass in an allocated one?

Reply:
`ASN1_item_d2i_bio` allocates the response for you, like the rest of our `d2i_*` functions. You can set the response to a value of `NULL` and it will be allocated a response if the function succeeds.